src/software/ai/hl/stp/play/shoot_or_pass/shoot_or_pass_play_fsm.cpp

```diff
@@ -47,24 +47,31 @@ void ShootOrPassPlayFSM::updateOffensivePositioningTactics(
 
 void ShootOrPassPlayFSM::lookForPass(const Update& event)
 {
-    PriorityTacticVector ret_tactics = {{attacker_tactic}, {}};
+    PriorityTacticVector ret_tactics = {{attacker_tactic, receiver_tactic}, {}};
 
     // only look for pass if there are more than 1 robots
     if (event.common.num_tactics > 1)
     {
         ZoneNamedN(_tracy_look_for_pass, "ShootOrPassPlayFSM: Look for pass", true);
+
         PassEvaluation<EighteenZoneId> pass_eval =
             pass_generator.generatePassEvaluation(event.common.world_ptr);
-        best_pass_and_score_so_far               = pass_eval.getBestPassOnField();
-        std::vector<EighteenZoneId> ranked_zones = pass_eval.rankZonesForReceiving(
-            event.common.world_ptr, event.common.world_ptr->ball().position());
+
+        auto best_pass_score_and_zone = pass_eval.getBestPassAndZoneOnField();
+        best_pass_zone                = best_pass_score_and_zone.first;
+        best_pass_and_score_so_far    = best_pass_score_and_zone.second;
+
+        auto ranked_zones = getBestOffensivePositions(pass_eval, event);
+
+        pass_eval = pass_generator.generatePassEvaluation(event.common.world_ptr);
```

Comment:
oh no! looks like the bug that was fixed in #3159 is reintroduced here :fearful: We're calling `generatePassEvaluation` twice!